fast64_internal/oot/scene/exporter/to_c/scene_header.py

```diff
@@ -1,208 +1,208 @@
 from .....utility import CData, indent
 from .....f3d.f3d_gbi import ScrollMethod, TextureExportSettings
 from ....oot_f3d_writer import OOTGfxFormatter
-from .scene_pathways import ootPathListToC
-from .actor import ootTransitionActorListToC, ootStartPositionListToC, ootEntranceListToC
-from .scene_commands import ootSceneCommandsToC
+from ....oot_level_classes import OOTScene, OOTLight
+from .scene_pathways import getPathData
+from .actor import getTransitionActorList, getSpawnActorList, getSpawnList
+from .scene_commands import getSceneCommandList
 
 
 ##################
 # Light Settings #
 ##################
-def ootVectorToC(vector):
-    return f"0x{vector[0]:02X}, 0x{vector[1]:02X}, 0x{vector[2]:02X}"
-
-
-def ootLightToC(light):
-    return (
-        indent
-        + "{ "
-        + ", ".join(
-            (
-                ootVectorToC(light.ambient),
-                ootVectorToC(light.diffuseDir0),
-                ootVectorToC(light.diffuse0),
-                ootVectorToC(light.diffuseDir1),
-                ootVectorToC(light.diffuse1),
-                ootVectorToC(light.fogColor),
-                light.getBlendFogShort(),
-                f"0x{light.fogFar:04X}",
-            )
-        )
-        + " },\n"
+def getColorValues(vector: tuple[int, int, int]):
+    return ", ".join(f"{v:5}" for v in vector)
+
+
+def getDirectionValues(vector: tuple[int, int, int]):
+    return ", ".join(f"{v - 0x100 if v > 0x7F else v:5}" for v in vector)
+
+
+def getLightSettingsEntry(light: OOTLight, lightMode: str, index: int):
+    vectors = [
+        (light.ambient, "Ambient Color", getColorValues),
+        (light.diffuseDir0, "Diffuse0 Direction", getDirectionValues),
+        (light.diffuse0, "Diffuse0 Color", getColorValues),
+        (light.diffuseDir1, "Diffuse1 Direction", getDirectionValues),
+        (light.diffuse1, "Diffuse1 Color", getColorValues),
+        (light.fogColor, "Fog Color", getColorValues),
+    ]
+
+    fogData = [
+        (light.getBlendFogNear(), "Blend Rate & Fog Near"),
+        (f"{light.fogFar}", "Fog Far"),
+    ]
+
+    lightDescs = ["Dawn", "Day", "Dusk", "Night"]
+    if lightMode == "false":
+        lightDesc = f"// {lightDescs[index]} Lighting\n"
+    else:
+        lightDesc = f"// {'Indoor' if lightMode == 'true' else 'Custom'} n°{index + 1} Lighting\n"
```

Comment:
This code throws an index error when exporting a kakariko import. I don't think I actually fully figured out how the lighting system worked, so the lighting mode enum representation might not be completely accurate. Kakariko has a `false` lighting mode but has 12 lights in it.

Reply:
that's odd, the lighting is set to custom, the condition shouldn't return true

Reply:
That member handles the custom property, so its going be equal to `"false"`.

Reply:
should be fixed now